odoo/addons/base/models/ir_model.py

```diff
@@ -1174,8 +1192,12 @@ def _instanciate_attrs(self, field_data):
             attrs['column1'] = field_data['column1'] or col1
             attrs['column2'] = field_data['column2'] or col2
             attrs['domain'] = safe_eval(field_data['domain'] or '[]')
-        elif field_data['ttype'] == 'monetary' and not self.pool.loaded:
-            return
+        elif field_data['ttype'] == 'monetary':
+            # be sure that custom monetary field are always instanciated
+            if not self.pool.loaded and \
+                not (field_data['currency_field'] and field_data['currency_field'].startswith('x_')):
```

Comment:
Without this, a tour testing the creation of a monetary field would fail because the pool was not loaded.
We did this patch with @rco-odoo and I must admit I didn't really have a clear understanding of the assumptions that led to this patch. Maybe @rco-odoo can clarify

Reply:
I must admit I don't remember the precise reason. This method is called before determining the fields on models. Therefore, at that point, there is no simple way to know for sure whether the currency field exists on the model. Except if the currency field is custom.

Maybe we should try again to make it work.

Reply:
The reason was that I wanted to keep the _former condition_, i.e., we ignore the monetary field if the currency field is not custom.  The new condition deals with the case where the currency field is custom, which is somehow a new use-case.